worlds/stardew_valley/rules.py

```diff
@@ -131,15 +131,19 @@ def set_tool_rules(logic: StardewLogic, multiworld, player, world_options: Stard
         MultiWorldRules.set_rule(tool_upgrade_location, logic.tool.has_tool(tool, previous))
 
 
-def set_building_rules(logic: StardewLogic, multiworld, player, world_options: StardewValleyOptions):
-    if not world_options.building_progression & BuildingProgression.option_progressive:
+def set_building_rules(logic: StardewLogic, multiworld, player, content: StardewContent):
+    building_progression = content.features.building_progression
+    if not building_progression.is_progressive:
         return
 
-    for building in locations.locations_by_tag[LocationTags.BUILDING_BLUEPRINT]:
-        if building.mod_name is not None and building.mod_name not in world_options.mods:
+    for building in content.farm_buildings.values():
+        if building.name in building_progression.starting_buildings:
```

Comment:
Making a note here from a voice chat conversation. This code here is an actual behavior change, that happens to not need any mod updates (because the mod is dynamic enough for it), but it is still a behavior change and a design philosophy change, to remove equivalent locations to some starting items.

We might need to think about a `start_inventory_and_skip` setting in the future to normalize this behavior and maybe give users some control over it. In the meantime, as long as we are consistent about it, I'm okay with moving forward with this game design. I'm not requesting a change, just making a written record of the choice.